lib/checkunusedvar.cpp

```diff
@@ -1489,8 +1494,35 @@ bool CheckUnusedVar::isRecordTypeWithoutSideEffects(const Type* type)
         if (f.argDef && Token::simpleMatch(f.argDef->link(), ") ="))
             continue; // ignore default/deleted constructors
         const bool emptyBody = (f.functionScope && Token::simpleMatch(f.functionScope->bodyStart, "{ }"));
-        const bool hasInitList = f.argDef && Token::simpleMatch(f.argDef->link(), ") :");
-        if (!emptyBody || hasInitList)
+
+        Token* nextToken = f.argDef->link();
+        if (Token::simpleMatch(nextToken, ")") && Token::simpleMatch(nextToken->next(), ":")) {
+            // validating initialization list
+            nextToken = nextToken->next(); // skip ":"
+            bool initListFinished = false;
+            while (!initListFinished)
```

Comment:
I would prefer that the loop walks 1 entry at a time..

    for (const Token *initListToken = nextToken; Token::Match(initListToken, "[:,] %var% [({]"); initListToken = initListToken->linkAt(2)->next()) {
        const Variable *variable = initListToken->next()->variable();
        if (variable && !isVariableWithoutSideEffects(*variable))
            return withoutSideEffects = false;
    }

It will then bailout quicker if something unexpected and strange happens. For instance if the code says `Fred(int) : ;` or something.

If the initlist says:   `:x(y)` do we want that this logic depends on only `x` or should it depend on `y` also?

Reply:
> If the initlist says: :x(y) do we want that this logic depends on only x or should it depend on y also?

My implementation handles both `x` and `y`. Code that you provided, works only with `x`.
I cannot provide and example right now where `y` should be also checked. But more checks are better than less checks, isn't it? =)

Reply:
I've just pushed new testcase, which checks that `y` type is validated also.

Reply:
I am not against that `y` is considered. but then I'd suggest an inner loop. It's good to have some syntax checking.

Reply:
I've refactored this loop according to your example. Please check.